alacritty_terminal/src/term/mod.rs

```diff
@@ -210,23 +210,29 @@ impl<'a, C> RenderableCellsIter<'a, C> {
         }
 
         let num_cols = self.grid.cols();
-        let cell = self.grid[&point];
+
+        // Since we can have non zero display offset, we must take points from a grid in buffer
+        // coordinates.
+        let buffer_point = self.grid.visible_to_buffer(point);
+        let cell = self.grid[buffer_point];
 
         // Check if wide char's spacers are selected.
         if cell.flags.contains(Flags::WIDE_CHAR) {
             let prev = point.sub(num_cols, 1);
```

Comment:
Why do we keep using the viewport point here, instead of using `sub_absolute`?

Reply:
The checks for selection are in visible coordinates. Also will `add_absolute`/`sub_absolute` clamp to viewport, I'm not sure that they will? Also, the point in `is_selected` is also in viewport, so you need to convert to viewport anyway.

Reply:
Sure, sounds good.